librz/bp/serialize_bp.c

```diff
@@ -0,0 +1,146 @@
+// SPDX-FileCopyrightText: 2021 DMaroo <dhruvsmaroo@gmail.com>
+// SPDX-License-Identifier: LPGL-3.0-only
+
+#include <rz_util/rz_serialize.h>
+#include <rz_bp.h>
+
+RZ_API void rz_serialize_bp_save(RZ_NONNULL Sdb *db, RZ_NONNULL RzBreakpoint *bp) {
+	rz_return_if_fail(db && bp);
+
+	RzListIter *iter;
+	RzBreakpointItem *bp_item;
+	rz_list_foreach (bp->bps, iter, bp_item) {
+		PJ *j = pj_new();
+		if (!j) {
+			return;
+		}
+		pj_o(j);
```

Comment:
@thestr4ng3r is it preferred to save as json or as sdb directly?

Reply:
My motivation for doing this was basically the code from `serialize_flag.c`. Much of the code in this file is similar to that of serializing flag zones.

Also, my 2 cents: Serializing directly as sdb will probably write every attribute of every breakpoint as a new key value pair. When we first serialize to JSON and then store as sdb, the data looks prettier and less cluttered.

Reply:
As a rule of thumb, an sdb entry should correspond to a single "object", whatever that means in the specific case. Attributes should be contained in the value as json.
So here, creating one sdb entry per breakpoint with all info inside of it as json is the best approach in my opinion.
You just have to make sure `bp_item->name` really is a unique identifier for the bp, I don't know the breakpoint code well enough to tell whether it is without checking.

Reply:
Yes. Names aren't a unique identifier for breakpoints. I'll use addresses instead.